libraries/ESP8266mDNS/src/LEAmDNS.h

```diff
@@ -462,6 +486,86 @@ class MDNSResponder {
     
 protected:
     /** STRUCTS **/
+    /**
+     * MDNSServiceInfo, used in application callbacks
+     */
+public:
+    struct MDNSServiceInfo
+    {
+    	MDNSServiceInfo(MDNSResponder& p_pM,MDNSResponder::hMDNSServiceQuery p_hS,uint32_t p_u32A)
+    	: p_pMDNSResponder(p_pM),
+    	  p_hServiceQuery(p_hS),
+		  p_u32AnswerIndex(p_u32A)
+    	{};
+    protected:
+    	MDNSResponder& p_pMDNSResponder;
+    	MDNSResponder::hMDNSServiceQuery p_hServiceQuery;
+    	uint32_t p_u32AnswerIndex;
+    public:
+    	const char* serviceDomain(){
+    		return p_pMDNSResponder.answerServiceDomain(p_hServiceQuery, p_u32AnswerIndex);
+    	};
+    	bool hostDomainAvailable()
+    	{
+          return (p_pMDNSResponder.hasAnswerHostDomain(p_hServiceQuery, p_u32AnswerIndex));
+    	}
+    	const char* hostDomain(){
+    		return (hostDomainAvailable()) ?
+    		   p_pMDNSResponder.answerHostDomain(p_hServiceQuery, p_u32AnswerIndex) : nullptr;
+    	};
+    	bool hostPortAvailable()
+    	{
+    		return (p_pMDNSResponder.hasAnswerPort(p_hServiceQuery, p_u32AnswerIndex));
+    	}
+    	uint16_t hostPort(){
+    		return (hostPortAvailable()) ?
+    		  p_pMDNSResponder.answerPort(p_hServiceQuery, p_u32AnswerIndex) : 0;
+    	};
+    	bool IP4AddressAvailable()
+    	{
+    		return (p_pMDNSResponder.hasAnswerIP4Address(p_hServiceQuery,p_u32AnswerIndex ));
+    	}
+    	std::vector<IPAddress> IP4Adresses(){
+    		std::vector<IPAddress> internalIP;
+    		if (IP4AddressAvailable()) {
+    			for (uint32_t u2 = 0; u2 < p_pMDNSResponder.answerIP4AddressCount(p_hServiceQuery, p_u32AnswerIndex); ++u2) {
+                  internalIP.push_back(p_pMDNSResponder.answerIP4Address(p_hServiceQuery, p_u32AnswerIndex, u2));
+    			}
+    		}
+    		return internalIP;
+    	};
+    	bool txtAvailable()
+    	{
+    		return (p_pMDNSResponder.hasAnswerTxts(p_hServiceQuery, p_u32AnswerIndex));
+    	}
+    	const char* strKeyValue (){
+    		return (txtAvailable()) ?
+    		  p_pMDNSResponder.answerTxts(p_hServiceQuery, p_u32AnswerIndex) : nullptr;
+    	};
+    	std::map<String,String> keyValues()
+		{
+    		std::map<String,String> tempKV;
+    		for (auto kv = p_pMDNSResponder._answerKeyvalue(p_hServiceQuery, p_u32AnswerIndex);kv != nullptr;kv = kv->m_pNext) {
+    			tempKV.insert(std::pair<String,String>(String(kv->m_pcKey),String(kv->m_pcValue)));

```

Comment:
If you can't move to a map of const char *, use of emplace(kv->m_pcKey, kv->m_pcValue) would be better here.

Reply:
Will do

Reply:
done